coordinator/src/services/translation/incoming/release/release-medication-request.ts

```diff
@@ -0,0 +1,297 @@
+  const lineItemQuantityTranslation = lineItemQuantity.quantity.translation
+  return {
+    value: lineItemQuantityTranslation._attributes.value,
+    unit: lineItemQuantityTranslation._attributes.displayName,
+    system: "http://snomed.info/sct",
+    code: lineItemQuantityTranslation._attributes.code
+  }
+}
+
+function createValidityPeriod(effectiveTime: Interval<Timestamp>): fhir.Period {
+  const validityPeriod: fhir.Period = {}
+  if (effectiveTime.low) {
+    validityPeriod.start = convertHL7V3DateToIsoDateString(effectiveTime.low)
+  }
+  if (effectiveTime.high) {
+    validityPeriod.end = convertHL7V3DateToIsoDateString(effectiveTime.high)
+  }
+  return validityPeriod
+}
+
+function createExpectedSupplyDuration(expectedUseTime: IntervalUnanchored): fhir.SimpleQuantity {
+  return {
+    unit: "day",
+    value: expectedUseTime.width._attributes.value,
+    system: "http://unitsofmeasure.org",
+    code: "d"
+  }
+}
+
+export function createDispenseRequest(
+  dispensingSitePreference: DispensingSitePreference,
+  lineItemQuantity: LineItemQuantity,
+  daysSupply: DaysSupply,
+  performer: Performer
+): fhir.MedicationRequestDispenseRequest {
+  const dispenseRequest: fhir.MedicationRequestDispenseRequest = {
+    extension: [
+      createPerformerSiteTypeExtension(dispensingSitePreference)
+    ],
+    quantity: createDispenseRequestQuantity(lineItemQuantity)
+  }
+  if (daysSupply?.effectiveTime?.low || daysSupply?.effectiveTime?.high) {
+    dispenseRequest.validityPeriod = createValidityPeriod(daysSupply.effectiveTime)
+  }
+  if (daysSupply?.expectedUseTime?.width) {
+    dispenseRequest.expectedSupplyDuration = createExpectedSupplyDuration(daysSupply.expectedUseTime)
+  }
+  if (performer) {
+    dispenseRequest.performer = createPerformer(performer.AgentOrgSDS.agentOrganizationSDS)
+  }
+  return dispenseRequest
+}
+
+function createPerformerSiteTypeExtension(dispensingSitePreference: DispensingSitePreference): fhir.CodingExtension {
+  return {
+    url: "https://fhir.nhs.uk/StructureDefinition/Extension-performerSiteType",
+    valueCoding: {
+      code: "https://fhir.nhs.uk/CodeSystem/dispensing-site-preference",
+      display: dispensingSitePreference.value._attributes.code
+    }
+  }
+}
+
+function createPerformer(performerOrganization: Organization): fhir.Performer {
+  return {
+    identifier: {
+      system: "https://fhir.nhs.uk/Id/ods-organization-code",
+      value: performerOrganization.id._attributes.extension
+    }
+  }
+}
+
+export function createGroupIdentifierFromPrescriptionIds(
+  prescriptionIds: [codes.GlobalIdentifier, codes.ShortFormPrescriptionIdentifier]
+): fhir.MedicationRequestGroupIdentifier {
+  const shortFormId = prescriptionIds[1]._attributes.extension
+  const longFormId = prescriptionIds[0]._attributes.root.toLowerCase()
+  return createGroupIdentifier(shortFormId, longFormId)
+}
+
+function createSubstitution() {
+  return {
+    allowedBoolean: false as const
+  }
+}
```

Comment:
This file is pretty big, is it worth splitting out some parts (e.g. the extension functions)?

Reply:
I don't think so, unless there's a reason. Everything in this file is to do with translating MedicationRequests. I don't think file size alone is a reason to split things up